plugins/sentinelone/komand_sentinelone/connection/connection.py

```diff
@@ -345,6 +346,66 @@ def get_events(self, params: dict, get_all_results: bool, event_type: str = None
 
         return insightconnect_plugin_runtime.helper.clean(self._call_api("GET", endpoint, params=params))
 
+    def update_analyst_verdict(self, incident_ids: list, analyst_verdict: str, _type: str) -> dict:
+        if _type == "threats":
+            endpoint = "threats/analyst-verdict"
+        else:
+            endpoint = "cloud-detection/alerts/analyst-verdict"
+
+        return self._call_api(
+            "POST",
+            endpoint,
+            {"filter": {"ids": incident_ids}, "data": {"analystVerdict": analyst_verdict}},
+        )
+
+    def update_incident_status(self, incident_ids: list, incident_status: str, _type: str) -> dict:
+        if _type == "threats":
+            endpoint = "threats/incident"
+        else:
+            endpoint = "cloud-detection/alerts/incident"
+
+        return self._call_api(
+            "POST",
+            endpoint,
+            {"filter": {"ids": incident_ids}, "data": {"incidentStatus": incident_status}},
+        )
+
+    def validate_incidents_exist(self, incident_ids: list, _type: str) -> list:
+        for incident_id in incident_ids:
+            response_data = self.get_incident(incident_id, _type).get("data")
+            if isinstance(response_data, list) and response_data.__len__() == 0:
+                self.logger.info(f"Incident {incident_id} was not found.")
+                incident_ids.remove(incident_id)
+
+        return incident_ids
+
+    def validate_incident_state(self, incident_ids: list, _type: str, new_state: str, attribute: str) -> list:
+        for incident_id in incident_ids:
+            response_data = self.get_incident(incident_id, _type).get("data")
+            for incident in response_data:
+                if _type == "threats":
+                    object_name = "threatInfo"
+                    resp_incident_id = incident.get("id")
+                else:
+                    object_name = "alertInfo"
+                    resp_incident_id = incident.get(object_name, {}).get("alertId")
+
+                attribute_name = "analystVerdict" if attribute == "analystVerdict" else "incidentStatus"
+                if resp_incident_id == incident_id and incident.get(object_name, {}).get(attribute_name) == new_state:
+                    self.logger.info(f"Incident {incident_id} has the {attribute_name} already set to {new_state}.")
+                    incident_ids.remove(incident_id)
+
+        return incident_ids
+
+    def get_incident(self, incident_id: str, _type: str) -> dict:
+        params = {"ids": [incident_id]}
+        if _type == "threats":
+            response = self.get_threats(params, api_version="2.1")
+        else:
+            response = self.get_alerts(params)
+
+        return response
+
```

Comment:
There are a lot of lengthy functions here. Could you leave some simple comments for each one describing what they do?

Reply:
I'll add a docstring to the functions. Both functions are used in a couple of actions so I would not recommend adding in-line comments in the action code.